src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Fdc3DesktopAgent.cs

```diff
@@ -27,14 +28,17 @@ public class Fdc3DesktopAgent : IHostedService
     private readonly List<UserChannel> _userChannels = new();
 
     private readonly ILoggerFactory _loggerFactory;
+    private readonly Fdc3Options _options;
     private readonly IMessageRouter _messageRouter;
-    private Fdc3DesktopAgentConfig? _config;
 
-    public Fdc3DesktopAgent(IMessageRouter messageRouter, ILoggerFactory? loggerFactory, Fdc3DesktopAgentConfig? config)
+    public Fdc3DesktopAgent(
+        IOptions<Fdc3Options> options, 
+        IMessageRouter messageRouter, 
+        ILoggerFactory? loggerFactory)
```

Comment:
Add default value (`ILoggerFactory? loggerFactory = null`) so that it can be optional in DI. Otherwise you'll get an exception instead of `null`.